### DISCUSSION TITLE
How can I check from code that a DataGrid is currently sorted?

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#14658] [Q&A] [answered]
I want to disable moving DataGrid rows up/down if the grid is currently sorted.
How can I check in code if a DataGrid is currently sorted?

**Comment by gehongyan** [comment, marked as answer]
You can wrap the `ItemsSource` of a `DataGrid` in a `DataGridCollectionView` so that you can access the `SortDescriptions` property through a reference to this `DataGridCollectionView` object when you need to retrieve sorting descriptions.

```csharp
// Wrap the ItemsSource in a DataGridCollectionView
DataGridCollectionView dataGridCollectionView = new(yourItemsSource);

// Assign the ItemsSource
YourDataGrid.ItemsSource = dataGridCollectionView;

// Access the SortDescriptions when needed
DataGridSortDescriptionCollection? sortDescriptions = (YourDataGrid.ItemsSource as DataGridCollectionView)?.SortDescriptions;
```

**Comment by wgraaf** [comment]
That works, thanks.
Wrapping ItemsSource is by far more preferable than resorting to reflection:
```
    var dataConnection = DataGrid.GetType().GetProperty("DataConnection", BindingFlags.Instance | BindingFlags.NonPublic)?.GetValue(DataGrid, null);
    _dataGridCollectionView = dataConnection?.GetType().GetProperty("DataSource")?.GetValue(dataConnection, null) as DataGridCollectionView;

```

The criterium for the grid being sorted becomes:

`DataGridCollectionView.SortDescriptions.Count > 0`

and subscribing to 

`DataGridCollectionView.SortDescriptions.CollectionChanged`

allows raising property changed notifications,
